pilz_control/include/pilz_control/pilz_joint_trajectory_controller_impl.h

```diff
@@ -178,6 +180,39 @@ triggerMovementToHoldPosition()
   JointTrajectoryController::setHoldPosition(this->time_data_.readFromRT()->uptime);
 }
 
+template <class SegmentImpl, class HardwareInterface>
+inline bool PilzJointTrajectoryController<SegmentImpl, HardwareInterface>::
+checkStates(const std::vector<double>& old_desired_position,
+            const std::vector<double>& new_desired_positioin,
+            const ros::Duration& period) const
+{
+  return (active_mode_ != Mode::HOLD
+          && !cartesian_speed_monitor_->cartesianSpeedIsBelowLimit(old_desired_position,
+                                                                   new_desired_positioin,
+                                                                   period.toSec(),
+                                                                   0.25 /*limit */));  // TODO: HSL
+}
+
+template <class SegmentImpl, class HardwareInterface>
+void PilzJointTrajectoryController<SegmentImpl, HardwareInterface>::
+reactToFailedStateCheck(const ros::Time& updated_uptime, const Trajectory& curr_traj)
+{
+  switchToHoldMode();
+  JointTrajectoryController::preemptActiveGoal();
```

Comment:
I would not rely on actionlib to meet the "realtime" requirements of the update cycle.
In other words this should be executed by another thread. Maybe also the other lines included in `reactToFailedStateCheck`. At least we have to take a close look at it.

Reply:
Done.